packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -946,11 +955,22 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 							);
 						}
 					}
-					await Promise.all(attachBlobsP);
+					await Promise.race([Promise.all(attachBlobsP), abortPromise]).catch(() => {});
 				}
 
 				for (const [id, entry] of this.pendingBlobs) {
+					if (stopWaitingAttachingSignal?.aborted && !entry.attached) {
+						this.mc.logger.sendTelemetryEvent({
+							eventName: "UnableToStashBlob",
+							id,
+						});
+						entry.unableToStash = true;
```

Comment:
why is this property needed?

Reply:
Such blob is orphan, meaning it was never attached/ not used in a dds. Used primarily to avoid sending the blob attach op in onUploadResolve if the blob is uploaded before closing the container.

Reply:
how is it different from being unattached